src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -126,14 +125,60 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
             echo "gfal-copy exit status: $EXIT_STATUS"
             if [[ $EXIT_STATUS != 0 ]]; then
                echo "ERROR: gfal-copy exited with $EXIT_STATUS"
+               echo "Actual command which failed: %s"
+               echo "Source PFN: %s"
+               echo "Target PFN %s"
                echo "Cleaning up failed file:"
                %s
             fi
             exit $EXIT_STATUS
-            """ % self.createRemoveFileCommand(targetPFN)
+            """ % copyCommand copyCommandDict['source'] copyCommandDict['destination'] self.createRemoveFileCommand(targetPFN)
 
         return result
 
+    def createDebuggingCommand(self, sourcePFN, targetPFN, options=None, checksums=None):
+        """
+        _createDebuggingCommand_
+        Report environment information about a failed gfal-copy command
+        
+        """
+        result = "#!/bin/bash\n"
+
+        copyCommandDict = {'checksum': '', 'options': '', 'source': '', 'destination': ''}
+
+        useChecksum = (checksums is not None and 'adler32' in checksums and not self.stageIn)
+
+        if not options:
+            options = ''
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--nochecksum', action='store_true')
+        args, unknown = parser.parse_known_args(options.split())
+
+        if not args.nochecksum:
+            if useChecksum:
+                checksums['adler32'] = "%08x" % int(checksums['adler32'], 16)
+                copyCommandDict['checksum'] = "-K adler32:%s" % checksums['adler32']
+            else:
+                copyCommandDict['checksum'] = "-K adler32"
+
+        copyCommandDict['options'] = ' '.join(unknown)
+
+        copyCommandDict['source'] = self.createFinalPFN(sourcePFN)
+        copyCommandDict['destination'] = self.createFinalPFN(targetPFN)
+
+        result += copyCommand
```

Comment:
I am afraid that we might give ourselves future problems with this attempt to copy the file once again in debug mode (i.e., outside of the actual stage out procedure).
I am uncertain, but perhaps we should not do any gfal copy at this stage, but only connect information about the node, the gfal version/binary, etc.

Reply:
Sounds reasonable, I will remove L170